C#/C#.sublime-syntax

```diff
@@ -1,278 +1,1634 @@
 %YAML 1.2
 ---
 # http://www.sublimetext.com/docs/3/syntax.html
-name: "C#"
+# https://github.com/gwenzek/csharpSyntax
+
+name: C#
 file_extensions:
   - cs
   - csx
 scope: source.cs
+
+
+variables:
+
+  bin_op: '(?:\+|->|-|\*|/|%|\|\||&&|\||&|\^|<<|>>|=>|<=|<|>=|>|==|!=|\?\?)'
+  unary_op: '(?:\+\+|--|-|~|!|&|\*)'
+  # name: '[A-Za-z_\p{L}][A-Za-z\p{L}_0-9]*'
+
+  # integers
+  hex_numeral: '0[xX][0-9a-fA-F]+'
+  decimal_numeral: '(0|[1-9][0-9]*)((?i:ul|lu|u|l))'
+
+  # floating point numbers
+  exponent: '[eE][+\-]?[0-9]+'
+  floating_point: '([0-9]+(?:\.[0-9]+)?(?:{{exponent}})?)([fFdDmM]?)'
+
+  # characters
+  unicode_char: '\\u[0-9a-fA-F]{,4}'
+  escaped_char: '(?:\\[btnfr"''\\]|{{unicode_char}}|\\x[0-9a-fA-F]{,4}||\\[0-9]{,3})'
+  # escaped_char: '\\{{unicode_char}}'
+  # char: '(?:{{escaped_char}}|.)'
+  # string: '(f|s|raw)?\"[^\"]*\"'
+  # literal: '({{number}}|{{char}}|{{string}})'
+
+  visibility: \b(public|private|protected|internal|protected\s+internal)\b
+  base_type: (?:bool|byte|sbyte|char|decimal|double|float|int|uint|long|ulong|short|ushort|object|string|void)
+  type_suffix: ((?:\[,*\]|\*|\?)*)
+  generic_declaration: \s*(<[^(={};<]*>)?\s*
+  reserved: '(?:abstract|as|base|break|case|catch|checked|class|const|continue|default|delegate|do|else|enum|event|explicit|extern|finally|fixed|for|foreach|goto|if|implicit|in|interface|internal|is|lock|namespace|new|null|operator|out|override|params|private|protected|public|readonly|ref|return|sealed|sizeof|stackalloc|static|string|struct|switch|this|throw|try|typeof|unchecked|unsafe|using|virtual|volatile|while)'
+  name: '(@{{reserved}}|@{{base_type}}|@var|{{name_normal}})'
+  # name: '([A-Za-z_][A-Za-z_0-9]*)'
+  # name_normal: '[A-Za-z_][A-Za-z_0-9]*'
+  # name_normal: '(?:[A-Za-z_\p{L}]|unicode_char)[A-Za-z_0-9\p{L}]*'
+  start_char: '(?:{{unicode_char}}|[A-Za-z_\p{L}])'
+  other_char: '(?:{{unicode_char}}|[A-Za-z_0-9\p{L}])'
+  name_normal: '{{start_char}}{{other_char}}*'
+  cap_name: '([A-Z]{{other_char}})'
```

Comment:
Instead of `[A-Z]` here you probably want `\p{Lu}`, unless unicode characters are not allowed. If that is the case, `\p{L}` probably shouldn't be used in the variables above.